client-app/src/pages/catalog/main.ts

```diff
@@ -52,7 +52,7 @@ new Vue({
 
 const criteria = new CartSearchCriteria();
 criteria.pageNumber = startPageNumber;
-criteria.pageSize = Number.MAX_SAFE_INTEGER;
+criteria.pageSize = dropdownMaxItems;
```

Comment:
dropdownMaxItems is using only for pageSize now, so why it is named dropdownMaxItems? )